### PR TITLE
Support aot_export torchbind op

### PR DESCRIPTION
**Pull request by ydwu4**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #124239
* __->__ #123370
* #123367

### CODE REVIEW

File: torch/_higher_order_ops/effects.py [hunk 1]
```diff
@@ -22,6 +22,10 @@ class _EffectType(Enum):
 }
 
 
+def register_side_effect_op(op: torch._ops.OpOverload) -> None:
```
Comment (angelayi) [inline torch/_higher_order_ops/effects.py:25]: haha i also added this [here](https://github.com/pytorch/pytorch/pull/122348/files#diff-f5c5ba29e62f89dc3d91301b1c5af37e0bc43b90fee45458dfddac0d217103c2R25). I guess we can see who lands first and go with whichever :>

Comment (ydwu4) [reply torch/_higher_order_ops/effects.py:25]: Sure!

File: test/export/test_torchbind.py [hunk 2]
```diff
@@ -713,6 +714,63 @@ def forward(self, arg0_1, arg1_1):
...
+            torch.empty(
+                0,
+            ).fill_(-1)
+        )
+        x = torch.ones(2, 3)
+
+        fake_mode = torch._subclasses.fake_tensor.FakeTensorMode()
+        fake_tq1 = torch._library.fake_class_registry.to_fake_obj(fake_mode, tq1)
+        fake_x = fake_mode.from_tensor(x)
+        gm = aot_export_module(mod, (fake_tq1, fake_x), trace_joint=False)[0]
```
Comment (zou3519) [inline test/export/test_torchbind.py:745]: to check, this doesn't use Dynamo?

**Review by zou3519 — APPROVED** [review]

File: test/export/test_torchbind.py (hunk 2 above)
Comment (ydwu4) [reply test/export/test_torchbind.py:775]: yeah!

**Pull request closed without merging**